plugins/repository-gcs/src/main/java/org/elasticsearch/repositories/gcs/GoogleCloudStorageService.java

```diff
@@ -180,6 +215,27 @@ StorageOptions createStorageOptions(final GoogleCloudStorageClientSettings clien
         return storageOptionsBuilder.build();
     }
 
+    @SuppressForbidden(reason = "ok to open connection here")
+    private static String getDefaultProjectId() throws IOException {
+        String metaHost = System.getenv("GCE_METADATA_HOST");
+        if (metaHost == null) {
+            metaHost = "metadata.google.internal";
+        }
+        URL url = new URL("http://" + metaHost + "/computeMetadata/v1/project/project-id");
+        HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+        connection.setConnectTimeout(5000);
```

Comment:
Can't we leave the defaults here?

Reply:
I've used the defaults that the SDK uses (and added a comment for this now)